airflow/cli/commands/task_command.py

```diff
@@ -75,6 +75,7 @@ def _run_task_by_executor(args, dag, ti):
             with create_session() as session:
```

Comment:
The create_session context manager commits the session before exiting the context

```
@contextlib.contextmanager
def create_session():
    """
    Contextmanager that will create and teardown a session.
    """
    session = settings.Session()
    try:
        yield session
        session.commit()
    except Exception:
        session.rollback()
        raise
    finally:
        session.close()
```